dvc/data/_progress.py

```diff
@@ -0,0 +1,32 @@
+from dvc.progress import Tqdm
+
+
+class QueryingProgress(Tqdm):
+    def __init__(self, iterable=None, total=None, name=None, phase="Querying"):
+        msg_part = "cache in " + f"'{name}'" if name else "remote cache"
+        msg_fmt = "{phase} " + msg_part
+
+        self._estimating_msg = msg_fmt.format(phase="Estimating size of")
+        self._listing_msg = msg_fmt.format(phase="Querying")
+        super().__init__(
+            iterable=iterable,
+            desc=msg_fmt.format(phase=phase),
+            total=total,
+            unit="files",
+            unit_scale=False,
+            bar_format=self.BAR_FMT_DEFAULT,
+        )
+
+    def callback(self, phase, *args):
+        total = args[0] if args else self.total
+        completed = args[1] if len(args) > 1 else self.n
```

Comment:
The progress callbacks take `completed`/`current`/`absolute` value rather than a relative/`to_increment` value. Maybe I am overengineering here, but in the future, we may want to throttle callbacks, so I felt like we should use `absolute` values in most places, instead of keeping callbacks guessing.